### DISCUSSION TITLE
Why assets  are not loading dynamically in build

### DISCUSSION THREAD

**Discussion by vinoddotcom** [nuxt/nuxt#17092] [Questions]
When this code running on development server then image loaded well, but when we run build then images is not loading 

```
<template>
  <div>
    <img :src="`/assets/img/${path}`" />
  </div>
</template>

<script setup>
const path = "dashboard.svg";

</script>
```

**Comment by Tch1b0** [comment]
The asset paths are needed on compile time.

The [assets guide](https://v3.nuxtjs.org/guide/features/assets/#assets-directory) says:
> Nuxt won't serve files in the assets/ directory at a static URL like /assets/my-file.png. If you need a static URL, use the [public/ directory](https://v3.nuxtjs.org/guide/features/assets/#public-directory)

So the path "/assets/img/dashboard.svg" might not exist anymore after compilation.

This means you can either do:
```html
<template>
  <div>
    <img src="/assets/img/dashboard.svg" />
  </div>
</template>
```

or

```html
<template>
  <div>
    <img :src="dashboardImage" />
  </div>
</template>

<script setup>
import dashboardImage from "/assets/img/dashboard.svg"
</script>
```

or maybe something else, but the full path must exist before the evaluation, so nothing relative like `"/assets/" + "my-file.png"`.

**Comment by vinoddotcom** [comment]
I am porting project from vue 3  to nuxt 3 in vue 3 we can import assets dynamically, but In nuxt  3 why we can't import assets  dynamically.

**Reply by Tch1b0** [reply]
Nuxt does a few things with assets when building, because of optimisation. So after building, the assets folder won't exist as it did before. But there is an easy solution: the `/public` directory will be served as it is. The directory and the files inside will remain the same even after building.

So if you would save the `dashboard.svg` in `/public/img/` you could use the code from above:
```html
<template>
  <div>
    <img :src="`/public/img/${path}`" />
  </div>
</template>

<script setup>
const path = "dashboard.svg";

</script>
```

**Reply by vinoddotcom** [reply]
thank you very much for providing best answer.
